spring-kafka/src/test/java/org/springframework/kafka/retrytopic/RetryTopicMultipleListenerIntegrationTests.java

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright 2021-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.retrytopic;
+
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.kafka.annotation.DltHandler;
+import org.springframework.kafka.annotation.EnableKafka;
+import org.springframework.kafka.annotation.KafkaListener;
+import org.springframework.kafka.annotation.RetryableTopic;
+import org.springframework.kafka.config.ConcurrentKafkaListenerContainerFactory;
+import org.springframework.kafka.core.*;
+import org.springframework.kafka.listener.ContainerProperties;
+import org.springframework.kafka.listener.DefaultErrorHandler;
+import org.springframework.kafka.support.KafkaHeaders;
+import org.springframework.kafka.test.EmbeddedKafkaBroker;
+import org.springframework.kafka.test.context.EmbeddedKafka;
+import org.springframework.messaging.handler.annotation.Header;
+import org.springframework.retry.annotation.Backoff;
+import org.springframework.stereotype.Component;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
+import org.springframework.util.backoff.FixedBackOff;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+
+/**
+ * @author Cenk Akin
+ * @since 3.0.3
+ */
+@SpringJUnitConfig
+@DirtiesContext
+@EmbeddedKafka(topics = {RetryTopicMultipleListenerIntegrationTests.FIRST_TOPIC,
+		RetryTopicMultipleListenerIntegrationTests.SECOND_TOPIC}, partitions = 1)
+public class RetryTopicMultipleListenerIntegrationTests {
```

Comment:
I got mainly inspired by [this test](https://github.com/spring-projects/spring-kafka/blob/0d1727da5bbd10cfff78845dfb0a265fa8f5d5e1/spring-kafka/src/test/java/org/springframework/kafka/retrytopic/RetryTopicSameContainerFactoryIntegrationTests.java#LL72C32-L72C32). Instead of adding a new test class, I can modify that class to make it cover this case or keep it as a separate class like in the PR. Or I can delete the test if you don't think it is worthwhile. It's up to you.

Reply:
Hi @cenkakin !

Not sure why do we need such a complex integration test just to verify that there is no conflict around `defaultRetryTopicKafkaTemplate` bean...
Right, I see that we need a second `@RetryableTopic` to make it fail after `gac.registerBean(RetryTopicBeanNames.DEFAULT_SCHEDULER_WRAPPER_BEAN_NAME,`, but probably we indeed can cover this use-case just adding extra config into that `RetryTopicSameContainerFactoryIntegrationTests` you mention before.

WDYT?

Thank you for contribution!

Reply:
Yup, it is what I can do. But we need to modify `Config` class in `RetryTopicSameContainerFactoryIntegrationTests` more to trigger the line that fails. 

As far as I understand from the test case intended to cover, it should be ok. Let me push another commit.